LrcParser.Tests/Parser/Lrc/Utils/LrcTimedTextUtilsTest.cs

```diff
@@ -12,26 +12,61 @@ public class LrcTimedTextUtilsTest
+    [TestCase(
+        "<00:06.84> Every <00:07.56> night <00:08.19> that <00:08.79> goes <00:09.59> between", 6840, "Every night that goes between",
+        new[] { "[0,start]:6840", "[6,start]:7560", "[12,start]:8190", "[17,start]:8790", "[22,start]:9590" }
+    )]
+    // Alternating time tags, unspaced
+    [TestCase(
+        "<00:06.84>Every<00:07.56>night<00:08.19>that<00:08.79>goes<00:09.59>between", 6840, "Everynightthatgoesbetween",
+        new[] { "[0,start]:6840", "[5,start]:7560", "[10,start]:8190", "[14,start]:8790", "[18,start]:9590" }
+    )]
+    [TestCase(
+        "Every<00:07.56>night<00:08.19>that<00:08.79>goes<00:09.59>between", 6840, "Everynightthatgoesbetween",
+        new[] { "[0,start]:6840", "[5,start]:7560", "[10,start]:8190", "[14,start]:8790", "[18,start]:9590" }
+    )]
+    // Alternating time tags, prefix spaced
+    [TestCase(
+        "<00:06.84> Every<00:07.56> night<00:08.19> that<00:08.79> goes<00:09.59> between", 6840, "Every night that goes between",
+        new[] { "[0,start]:6840", "[6,start]:7560", "[12,start]:8190", "[17,start]:8790", "[22,start]:9590" }
+    )]
+    [TestCase(
+        "Every<00:07.56> night<00:08.19> that<00:08.79> goes<00:09.59> between", 6840, "Every night that goes between",
+        new[] { "[0,start]:6840", "[6,start]:7560", "[12,start]:8190", "[17,start]:8790", "[22,start]:9590" }
+    )]
+    // Alternating time tags, postfix spaced
+    [TestCase(
+        "<00:06.84>Every <00:07.56>night <00:08.19>that <00:08.79>goes <00:09.59>between", 6840, "Every night that goes between",
+        new[] { "[0,start]:6840", "[6,start]:7560", "[12,start]:8190", "[17,start]:8790", "[22,start]:9590" }
+    )]
+    [TestCase(
+        "Every <00:07.56>night <00:08.19>that <00:08.79>goes <00:09.59>between", 6840, "Every night that goes between",
+        new[] { "[0,start]:6840", "[6,start]:7560", "[12,start]:8190", "[17,start]:8790", "[22,start]:9590" }
+    )]
+    // Multiple empty tags
+    [TestCase(
+        "<00:51.00> <01:29.99><01:48.29>  <02:31.00> <02:41.99>You gotta fight !", 0,
+        "You gotta fight !", new[] { "[0,start]:161990" }
+    )]
+    public void TestDecode(string text, int lineStartTime, string expectedText, string[] expectedTimeTags)
     {
-        var (actualText, actualTimeTags) = LrcTimedTextUtils.TimedTextToObject(text);
-
-        Assert.That(actualText, Is.EqualTo(expectedText));
-        Assert.That(actualTimeTags, Is.EqualTo(TestCaseTagHelper.ParseTimeTags(expectedTimeTags)));
-    }
-
-    [TestCase("<00:51.00><01:29.99><01:48.29><02:31.00><02:41.99>You gotta fight !", "You gotta fight !", new[] { "[0,start]:51000" })] // decode with invalid format.
-    public void TestDecodeWithInvalidFormat(string text, string expectedText, string[] expectedTimeTags)
-    {
-        var (actualText, actualTimeTags) = LrcTimedTextUtils.TimedTextToObject(text);
+        var (actualText, actualTimeTags) = LrcTimedTextUtils.TimedTextToObject(text, lineStartTime);

```

Comment:
Have no idea why you remove this test case.

Reply:
That's the one I moved above, since the parsing itself passes fine. Thus, I don't consider it invalid.

Reply:
Not really :smiling_face_with_tear:
However, the approach I chose seems logical to me:
`<00:51.00><01:29.99><01:48.29><02:31.00><02:41.99>You gotta fight !`
There are four segments that have a time defined, but no content. Thus, they don't need to be extracted and will be skipped. The last segment isn't empty, and will have a start time of `02:41.99`.